packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/PersonalInformationStep/PersonalInformationStep.tsx

```diff
@@ -22,9 +31,31 @@ const PersonalInformationStep = ({
     useUpdateProfileByAddressMutation();
   const [emailBoundCheckboxKey, setEmailBoundCheckboxKey] = useState(1);
 
+  const [currentUsername, setCurrentUsername] = useState('');
+  const debouncedSearchTerm = useDebounce<string>(currentUsername, 500);
+
+  const { data: profiles, isLoading: isCheckingUsernameUniqueness } =
+    useSearchProfilesQuery({
+      limit: 1000,
```

Comment:
do we  need such a high limit? technically we should throw an error if at least 1 result is returned so the limit could be `1`? 

not sure how this works (can't build the project) but in this case maybe we should do the validation on submit, not on every input change 🤔

Reply:
hm, submit makes sense too, I will update that.